app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -20,6 +20,16 @@ def current_repositories(environment_id = nil, content_view_id = nil)
         repo_ids = repos_on_capsule.map(&:name)
         katello_repos.select { |repo| repo_ids.include? repo.pulp_id }
       end
+
+      def orphaned_repositories
+        katello_pulp_ids = Katello::Repository.all.select { |repo| smart_proxy.pulp3_support?(repo) }.map(&:pulp_id)
+        repos_on_capsule = ::Katello::Pulp3::Repository.new(nil, smart_proxy).list({})
+        repos_on_capsule.reject { |capsule_repo| katello_pulp_ids.include? capsule_repo.name }
+      end
+
+      def delete_orphaned_repositories
+        orphaned_repositories.map { |repo| ::Katello::Pulp3::Repository.new(nil, smart_proxy).repositories_api.delete(repo._href) }
```

Comment:
I'd probably name both of these to be specific for 'mirror_proxies'

Reply:
Also, talking with @jturel he asked a question about if the repos we are clearing out are constrained by environment, etc. I didn't think we were filtering by any criteria, but maybe that's incorrect?

Reply:
yep, was just typing that now!  there's a method in pulp/smart_proxy_repository.rb  current_repositories that should return the repos that belong on the capsule.  I believe its only used for orphan cleanup, in which case i don't think we need to differentiate between pulp2 or pulp3.  Meaning its okay if we're looking a list that contains both.

Reply:
and you may want to move that method to smart_proxy_extensions.rb since its really not pulp2/3 related.

Reply:
> you may want to move that method 

I am not certain which method you meant here.

Reply:
the 'current_repositories' method in pulp/smart_proxy_repository.rb